nixos/modules/services/misc/n8n.nix

```diff
@@ -0,0 +1,432 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.n8n;
+
+  baseConfig = builtins.removeAttrs cfg.config [ "listenAddress" "sslKeyFile" "sslCertFile" ] // {
+    listen_address = cfg.config.listenAddress;
+    ssl_key = cfg.config.sslKeyFile;
+    ssl_cert = cfg.config.sslCertFile;
+  };
+  filteredConfig =
+    filterAttrsRecursive (n: v: v != {})
+    (filterAttrsRecursive (n: v: v != {})
+      (filterAttrsRecursive (n: v: v != null && v != {} && v != [] && v != "")
+        baseConfig
+      )
+    );
+  configFile = pkgs.writeText "n8n.conf" (builtins.toJSON filteredConfig);
+in
+{
+  options.services.n8n = {
+
+    enable = mkEnableOption "n8n server.";
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Open ports in the firewall for the n8n web interface.";
+    };
+
+    package = mkOption {
+      type = types.path;
+      default = pkgs.n8n;
+      defaultText = "pkgs.n8n";
+      description = "Package containing the n8n program.";
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "n8n";
+      description = "User account under which n8n runs.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "n8n";
+      description = "Group under which n8n runs.";
+    };
+
+    config = {
```

Comment:
Have you looked at [RFC42](https://github.com/NixOS/rfcs/pull/42) yet?

Reply:
No, added to the list

Reply:
So basically, I rename my "config" key to a single empty option, link to the documentation, and use a lib to build the json file?

Is there a general policy for the default?

Reply:
> So basically, I rename my "config" key to a single empty option, link to the documentation, and use a lib to build the json file?

Yes. If you don't think this provides enough value to the sysadmin you could utilize a [freeform module option](I think in this case specifically you would benefit from using a [freeform module option](https://github.com/NixOS/nixpkgs/pull/82743) to provide type checking and validation to your `settings` option.

Please don't hesitate to ask for any help on this PR :smile:

Reply:
Ok I see in the config part, instead of the options